internal/config/migration/multi_account.go

```diff
@@ -0,0 +1,215 @@
+package migration
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/internal/keyring"
+	ghAPI "github.com/cli/go-gh/v2/pkg/api"
+	"github.com/cli/go-gh/v2/pkg/config"
+)
+
+type CowardlyRefusalError struct {
+	err error
+}
+
+func (e CowardlyRefusalError) Error() string {
+	// Consider whether we should add a call to action here like "open an issue with the contents of your redacted hosts.yml"
+	return fmt.Sprintf("cowardly refusing to continue with multi account migration: %s", e.err.Error())
+}
+
+var hostsKey = []string{"hosts"}
+
+// This migration exists to take a hosts section of the following structure:
+//
+//	github.com:
+//	  user: williammartin
+//	  git_protocol: https
+//	  editor: vim
+//	github.localhost:
+//	  user: monalisa
+//	  git_protocol: https
+//    oauth_token: xyz
+//
+// We want this to migrate to something like:
+//
+// github.com:
+//	 user: williammartin
+//	 git_protocol: https
+//	 editor: vim
+//	 users:
+//	   williammartin:
+//	     git_protocol: https
+//	     editor: vim
+//
+// github.localhost:
+//	 user: monalisa
+//	 git_protocol: https
+//   oauth_token: xyz
+//   users:
+//	   monalisa:
+//	     git_protocol: https
+//	     oauth_token: xyz
+//
+// The reason for this is that we can then add new users under a host.
+// Note that we are only copying the config under a new users key, and
+// under a specific user. The original config is left alone. This is to
+// allow forward compatability for older versions of gh and also to avoid
+// breaking existing users of go-gh which looks at a specific location
+// in the config for oauth tokens that are stored insecurely.
+
+type MultiAccount struct {
+	// Allow injecting a transport layer in tests.
+	Transport http.RoundTripper
+}
+
+func (m MultiAccount) PreVersion() string {
+	// It is expected that there is no version key since this migration
+	// introduces it.
+	return ""
+}
+
+func (m MultiAccount) PostVersion() string {
+	return "1"
+}
```

Comment:
I imagine this is arbitrary, but how would you feel about making this a `2` or `1.1`?

At this point, I don't know how many iterations this configuration has gone through but it isn't quite a `1.0`

Reply:
I have no strong feelings. I decided on `1` simply because it had never been versioned before and therefore everything else conceptually was `0.x`. I'm happy to change this if you feel strongly.

Reply:
Eh, don't want to create additional work as you say it is murky at best.  Thanks for entertaining the question 🙇